tests/test_apply.py

```diff
@@ -183,24 +161,25 @@ def test_invalid_qubit(self, init_state, device):
         state = init_state(1)
 
         with pytest.raises(qml.QuantumFunctionError, match="The QubitStateVector operation is not supported on the unitary simulator backend"):
-            dev.apply("QubitStateVector", [0], [state])
+            dev.apply([qml.QubitStateVector(state, wires=[0])])
 
 @pytest.mark.parametrize("shots", [8192])
 @pytest.mark.parametrize("analytic", [False])
 @pytest.mark.usefixtures("skip_unitary")
-class TestHardwareApply:
```

Comment:
I am a bit confused...so are Hardware backends only tested with ``analytic==False`` and State backends with ``analytic==True``? What about the other two cases?

Reply:
Yes, this caused a bit of confusion for me as well initially (thanks to @josh146 who helped me out with spotting some cases that were not covered!).

What we had:
* `TestStateApply`: setting `analytic=True` and using `device` fixture that checks **all** types of backends (so not just the state backends)
* `TestHardwareApply`: setting `analytic=False` and using `device` fixture that checks **all** types of backends (so not just the hardware backends)

My confusion came from the fast that the names for the classes did not reflect what was going on in the tests, so I've renamed them and adjusted the docstrings. Let me know, however, if you think something is still amiss.

Reply:
Cool, all good!